### DISCUSSION TITLE
Dinamically change theme

### DISCUSSION THREAD

**Discussion by xylish7** [heroui-inc/heroui#2988] [Help]
Is there a way to dynamically change the them inside an application and save the user preferences. for example, in local storage? Something like the [ThemeProvider](https://mui.com/material-ui/customization/theming/#themeprovider) from MUI where you can pass a dinamically created theme.

**Comment by fillobotto** [comment]
Up for this

**Comment by wingkwong** [comment]
You can customize your theme in tailwind config and use `use-theme` hook to change based on user preference. 

see

- https://www.heroui.com/docs/customization/customize-theme
- https://github.com/heroui-inc/heroui/tree/canary/packages/hooks/use-theme

**Reply by fillobotto** [reply]
This assumes, however, that the theme variant is hardcoded in a configuration file.

My use case, instead, is that the theme color values are retrieved via an API and applied dynamically
